File: typescript/src/fib.ts
type MemoObject = {
	[key: number]: number
}

function fib(index: number, obj: MemoObject = {}): number {
	if (index <= 2) {
		return 1
	} else if (index in obj) {
		return obj[index]
	}

	return obj[index] = fib(index - 1, obj) + fib(index - 2, obj)
}

console.log(
	fib(12) // 144
)

console.log(
	fib(13) // 233
)

File: python/src/fib.py
def fib(index: int):
	assert type(index) == int, f"{index} must be an integer"

	if index <= 2:
		return 1

	return fib(index - 1) + fib(index - 2)

if __name__ == "__main__":
	print(
		fib(12) # 144
	)

	print(
		fib(13) # 233
	)